modules/dnn/include/opencv2/dnn/utils/npy_blob.hpp

```diff
@@ -0,0 +1,25 @@
+#ifndef OPENCV_DNN_NPY_BLOB_HPP
+#define OPENCV_DNN_NPY_BLOB_HPP
+
+#include <opencv2/core.hpp>
+#include <string>
+
+namespace cv {
+namespace dnn {
+
+/** @brief Reads a NumPy array (*.npy) from file and returns it as a cv::Mat.
+ *  @param path Path to the .npy file.
+ *  @return cv::Mat containing the loaded array.
+ *
+ *  The function supports the following NumPy data types:
+ *  - <f4 (float32)
+ *  - <i4 (int32)
+ *  - <i8 (int64)
+ *
+ *  Note: Only C-style ordering (fortran_order=False) is supported.
+ */
+CV_EXPORTS_W Mat blobFromNPY(const String& path);
```

Comment:
Are these implementation and API secure (safe enough to handle malicious input)?

At least there is no any protection against OOM DDOS attack.